rfcs/QueryLevelNullability.md

```diff
@@ -1,100 +1,158 @@
-write `Int("5")!`. If you do that, an exception will be thrown if the statement would return `null`.
+referred to as the "force unwrap operator".
 
-In GraphQL, the `!` operator will act similarly. In the case that `name` does not exist, the query will return an
-error to the client rather than any data.
+In Swift, for example, you can cast a string to an integer with `Int("5")` but the string being cast may not be a valid
+number, so that statement will evaluate to `null` rather than an integer if the string cannot be cast to an integer. If
+you want to ensure that the statement does not return `null` you can instead write `Int("5")!`. If you do that, an
+exception will be thrown if the statement would evaluate to `null`.
 
-On platforms where codegen is used, clients have full control over the nullability of the properties on the
-generated types. Since nullability is expressed in the query rather than the schema, it's flexible enough to accommodate
-various use-cases (e.g., where the business `name` _is_ allowed to be nullable).
+In GraphQL, the `!` operator will act similarly. In the case that `name` does not exist, the query will return an error
+to the client rather than any data.
 
 ### `!`
 
 We have chosen `!` because `!` is already being used in the GraphQL spec to indicate that a field is non-nullable.
 Incidentally the same precedent exists in Swift (`!`) and Kotlin (`!!`) which both use similar syntax to indicate
-"throw an exception if this value is `null`". 
+"throw an exception if this value is `null`".
 
 ### Use cases
 
 #### When a field is necessary to the function of the client
-There are times where a field can be `null`, but a feature that queries for the field will not function
-if it is `null`. For example if you are trying to render an information page for a movie, you won't
-be able to do that if the name field of the movie is missing. In that case it would be preferable
-to fail as early as possible.
+
+Expressing nullability in the query, as opposed to the schema, offers the client more flexibility and control over
+whether or not an error is thrown.
+
+There are cases where a field is `nullable`, but a feature that fetches the field will not function if it is `null`.
+For example if you are trying to render an information page for a movie, you won't be able to do that if the name field
+of the movie is missing. In that case it would be preferable to fail as early as possible.
+
+According to the official GraphQL best practice, this field should be non-nullable:
+
+> When designing a GraphQL schema, it's important to keep in mind all the problems that could go wrong and if "null" is
+> an appropriate value for a failed field. Typically it is, but occasionally, it's not. In those cases, use non-null
+> types to make that guarantee.
+
+However, the same field may not be "vital" for every feature in the application that fetches it. Marking the field as
```

Comment:
I extended this example to include a counterexample, since the best practice has guidance for the case where some field isn't useful when `null`. I think this better illustrates our dilemma and explains why that guidance isn't suitable for us, lmk whatcha think

Reply:
Great idea! Looks good